crates/sparrow-core/src/read_config.rs

```diff
@@ -0,0 +1,24 @@
+use arrow::datatypes::SchemaRef;
+
+/// Defines the configuration for a read from a source.
+#[derive(Debug)]
+pub struct ReadConfig {
+    /// The schema of the requested data.
+    ///
+    /// The actual schema of the input may have additional columns.
+    pub projected_schema: SchemaRef,
+    /// If true, the read will act as an unbounded source and continue reading
+    /// as new data is added. It is on the consumer to close the channel.
+    ///
+    /// If false, the read will act as a bounded source, and stop once the set
+    /// of data available at the time of the read has been processed.
+    pub keep_open: bool,
+    /// Optional timestamp in nanos at which to start reading.
+    ///
+    /// Defaults to the earliest available timestamp.
+    pub start_time: Option<i64>,
+    /// Optional timestamp in nanos at which to end reading.
+    ///
+    /// Defaults to reading until the source is closed.
+    pub end_time: Option<i64>,
```

Comment:
I wonder if it would be clearer to combine `keep_open` and `end_time`:

```
pub enum End {
  Never,
  AtTime(i64),
  AtEndOfCurrentData,
```

Reply:
add to doc and comment about keepopen/endtime